### PR TITLE
CoordinatePosition handles degenerate LineString

### LINKED ISSUE #1482: Panic in Relate when using Degenerate LineString
## Describe the bug
`Relate::relate` Panicked on an edge case

## Library Version
`geo` 0.32.0

## Expected Behaviour
Degenerate LineString.relate(&Polygon) should not Panic

## To Reproduce
```rust

#[test]
    fn test_degenerate_linestring() {
        let pt: Point<f64> = wkt! {POINT(1 1)}.convert();
        let poly: Polygon<f64> = wkt! {RECT(0 0, 1 1)}.to_polygon().convert();
        let ls1: LineString<f64> = wkt! {LINESTRING(1 1)}.convert();
        // this is ok
         ls1.relate(&pt).is_intersects();
         // this is ok
         ls1.relate(&ls1).is_intersects();
         // this panics
         ls1.relate(&poly);
    }

```

## System (please complete the following information)
 - OS: macOS
 - OS version if possible

## Additional Context
Stack Trace seems quite clear on where the problem stems from

```
thread 'algorithm::intersects::line_string::test::test_degenerate_linestring' (31304056) panicked at geo/src/algorithm/coordinate_position.rs:141:13:
invalid line string with less than 2 coords
stack backtrace:
   0: __rustc::rust_begin_unwind
             at /rustc/ed61e7d7e242494fb7057f2657300d9e77bb4fcb/library/std/src/panicking.rs:698:5
   1: core::panicking::panic_fmt
             at /rustc/ed61e7d7e242494fb7057f2657300d9e77bb4fcb/library/core/src/panicking.rs:75:14
   2: <geo_types::geometry::line_string::LineString<T> as geo::algorithm::coordinate_position::CoordinatePosition>::calculate_coordinate_position
             at ./src/algorithm/coordinate_position.rs:141:13
   3: <geo::geometry_cow::GeometryCow<T> as geo::algorithm::coordinate_position::CoordinatePosition>::calculate_coordinate_position
             at ./src/types.rs:146:51
   4: geo::algorithm::coordinate_position::CoordinatePosition::coordinate_position
             at ./src/algorithm/coordinate_position.rs:42:14
   5: geo::algorithm::relate::relate_operation::RelateOperation<F,BBOX1,BBOX2>::label_isolated_node
             at ./src/algorithm/relate/relate_operation.rs:376:33
   6: geo::algorithm::relate::relate_operation::RelateOperation<F,BBOX1,BBOX2>::label_isolated_nodes
             at ./src/algorithm/relate/relate_operation.rs:363:21
   7: geo::algorithm::relate::relate_operation::RelateOperation<F,BBOX1,BBOX2>::compute_intersection_matrix
             at ./src/algorithm/relate/relate_operation.rs:105:14
   8: geo::algorithm::relate::Relate::relate
             at ./src/algorithm/relate/mod.rs:74:43
   9: geo::algorithm::intersects::line_string::test::test_degenerate_linestring
             at ./src/algorithm/intersects/line_string.rs:104:14
  10: geo::algorithm::intersects::line_string::test::test_degenerate_linestring::{{closure}}
             at ./src/algorithm/intersects/line_string.rs:95:36
  11: core::ops::function::FnOnce::call_once
             at /Users/seed1091/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/function.rs:250:5
  12: core::ops::function::FnOnce::call_once
             at /rustc/ed61e7d7e242494fb7057f2657300d9e77bb4fcb/library/core/src/ops/function.rs:250:5
```

### PR DESCRIPTION
**Pull request by michaelkirk**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---
FIXES #1482 

~~Draft because: depends on #1481, merge that first.~~ ✅ merged

### CODE REVIEW

File: geo/src/algorithm/coordinate_position.rs [hunk 1]
```diff
@@ -577,62 +584,76 @@ mod test {
...
+        #[test]
+        fn test_simple_line_string() {
+            let line_string: LineString = wkt!(LINESTRING(0 0,1 1,2 0,3 0)).convert();
 
-        let start = Coord::zero();
-        assert_eq!(
-            line_string.coordinate_position(&start),
-            CoordPos::OnBoundary
-        );
+            let start = Coord::zero();
```
Comment (michaelkirk) [inline geo/src/algorithm/coordinate_position.rs:593]: A lot of these lines are just indentation changes, though I did add some new test cases. Consider reviewing with whitespace changes hidden.
